pkg/storaged/crypto-keyslots.jsx

```diff
@@ -355,67 +342,68 @@ function edit_tang_adv(client, block, key, url, adv, passphrase) {
-                    </h3>
-                    <p>{ fmt_to_fragments(_("Passphrase removal may prevent unlocking $0."), <b>{dev}</b>) }</p>
-
-                    <div name="remove-passphrase" className="progressive-disclosure ct-form">
-                        <div className="form-group">
-                            <label>
-                                <input type="radio" checked={val !== false}
-                                       autoFocus
-                                       onChange={event => change("")} />
-                                {_("Confirm removal with passphrase")}
-                            </label>
-                            <input className="form-control" type="password" hidden={val === false}
-                                   value={val} onChange={event => change(event.target.value)} />
-                        </div>
-                        <div className="form-group">
-                            <label>
-                                <input type="radio" checked={val === false}
-                                       onChange={event => change(false)} />
-                                { fmt_to_fragments(_("Force remove passphrase in $0"), <b>{key_slot}</b>) }
-                            </label>
-                            <p className="slot-warning" hidden={val !== false}>
-                                Removing a passphrase without confirmation can be dangerous.
-                            </p>
-                        </div>
-                    </div>
-                </div>
-            );
-        }
-    };
-};
+        this.onRemove = this.onRemove.bind(this);
+    }
 
-function remove_passphrase_dialog(block, key) {
-    dialog_open({
-        Title: _("Remove passphrase"),
-        Fields: [
-            RemovePassphraseField("passphrase", key, block_name(block))
-        ],
-        Action: {
-            DangerButton: true,
-            Title: _("Remove"),
-            action: function (vals) {
-                if (vals.passphrase === false)
-                    return slot_remove(block, key.slot);
-                else
-                    return passphrase_remove(block, vals.passphrase);
-            }
+    onRemove() {
+        this.setState({ pending: true });
+        const dev = decode_filename(this.props.block.Device);
+        const opts = { superuser: true, err: "message" };
+        const cmd = ["cryptsetup", "luksKillSlot", dev, this.props.slot.toString()];
+        if (this.state.force) {
+            cmd.splice(2, 0, "-q");
+            opts.pty = true;
         }
-    });
+
+        const spawn = cockpit.spawn(cmd, opts);
+        if (!this.state.force)
+            spawn.input(this.state.passphrase + "\n");
+
+        spawn.then(this.props.onClose)
+                .catch(e => this.setState({ error: e.message }))
+                .always(() => this.setState({ pending: false }));
+    }
+
+    render() {
+        return (
+            <Modal
```

Comment:
I don't think we should avoid the dialog abstraction in dialog.jsx unless it makes things really really hard. Rather, we should extend the abstraction to make totally custom layout possible. I'll give that a shot.

Reply:
Here is what I have so far: https://github.com/cockpit-project/cockpit/compare/master...mvollmer:storage-new-remove-key-layout-experiment

What do you think?

Reply:
Good call! I've picked the idea and pushed it here. Content of dialogs looks the same, just the dialog itself looks as rest of the dialogs.
Thanks!